### PR TITLE
Classify genotypes for plate 4

### PR DESCRIPTION
**Pull request by MattsonCam**
In this pr, two logistic regression models are trained on both the shuffled and unshuffled plate 4 data to classify nf1 genotypes. Please let me know if you have any questions or suggestions.

### CODE REVIEW

File: 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+# ## Split the Data
+
+# In[9]:
+
+
+test_frac = 0.2
+val_frac = 0.15
+
+X_train, X_test, y_train, y_test = train_test_split(featdf, platedf[gene_column], test_size = test_frac, random_state=0, shuffle=True, stratify=platedf[gene_column])
```
Comment (MikeLippincott) [inline 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py:127]: Consider making the `shuffle=True` arg modular so you can run this for both unshuffled and shuffled data

File: 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+# ## Shuffle the training data for the shuffled model
+
+# In[10]:
+
+
+X_train_shuf = X_train.copy()
+
+rng = np.random.default_rng(0)
+
+for column in X_train_shuf.columns:
```
Comment (MikeLippincott) [inline 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py:139]: Not 100% sure what is happining here. Consider adding more in line documentation. Also did  you mean to shuffle the data twice?

File: 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+# In[11]:
+
+
+# Define the hyperparameter search space
+param_dist = {
+    'C': uniform(loc=0.5, scale=1.5),
+    "l1_ratio": uniform(loc=0, scale=1),
+}
+
+# Create a Logistic Regression model
```
Comment (MikeLippincott) [inline 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py:156]: consider adding inline citation for Roshan's phenotypic model if this is where you got this code from.
"Code adapted from:...."

**Review by MikeLippincott — APPROVED** [review]
Nice PR. Looks like you may have shuffled the data twice. Consider double checking this change prior to merging. Other than that. LGTM.

File: 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py (hunk 1 above)
Comment (MattsonCam) [reply 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py:127]: We talked about this one

File: 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py (hunk 2 above)
Comment (MattsonCam) [reply 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py:139]: Oh, just talked about this too.

File: 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py (hunk 3 above)
Comment (MattsonCam) [reply 1.train_models/log_reg_plates_cp_fs_data/plate_4/nbconverted/plate_4_classify_genotype.py:156]: I don't remember if it was exactly the code, or just the same approach, but good to think about.

**Comment by MattsonCam** [comment]
Thanks @MikeLippincott,  Merging now!
